lib/neo4j/shared/query_factory.rb

```diff
@@ -89,6 +97,12 @@ def create_query
                 .params(:"#{identifier}_create_props" => create_props)
     end
 
+    def create_query!
+      return match_query if graph_object.persisted?
```

Comment:
So this line:
```
return match_query if graph_object.persisted?
```
exists in the, already existing, `create_query` method. `match_query` simply matches on,
```
"(#{graph_object.from_node_identifier})-[#{identifier}]->()"
```
It's unclear to me why you can get away without also specifying the `to_node`. In my testing though, it didn't seem as if `graph_object.persisted?` ever `== true` while setting an association, so I decided to just leave it as is. in `create_query!`